engine/runtime-with-polyglot/src/test/java/org/enso/interpreter/test/ConversionMethodTests.java

```diff
@@ -101,4 +119,26 @@ public void testDispatchOnJSDateTime() {
     Value res = evalModule(ctx, src);
     assertEquals(7, res.asInt());
   }
+
+  @Test
+  public void testAmbiguousConversionStrictUnused() {
+    String src = """      
+       type Foo
+          Mk_Foo data
+       type Bar
+          Mk_Bar x
+       
+       Foo.from (that:Bar) = Foo.Mk_Foo that.x+100
+       Foo.from (that:Bar) = Foo.Mk_Foo that.x+1000
+       
+       main = 42
+       """;
+    try {
+      Value res = evalModule(ctx, src);
+      fail("Expected an exception, but got " + res);
+    } catch (Exception e) {
+      assertEquals("Compilation aborted due to errors.", e.getMessage());
+      MatcherAssert.assertThat(getStdOut(), Matchers.containsString("Unnamed:7:1: error: Ambiguous conversion: Foo.from Bar is defined multiple times in this module."));
```

Comment:
minor: I believe static methods from `org.hamcrest.MatcherAssert` should follow the same convention as methods from `org.junit.Assert`. That is, they should be directly imported, so that one can write jus `assertThat(..., containsString("...")`.

Reply:
Simple `assertNotEquals(getStdOut().indexOf(....), -1)` would be my favorite alternative.

Reply:
We already have the dependency available, so I don't see why we shouldn't use it - using the specialized matcher yields much better error messages than '-1 did equal -1'.